### PR TITLE
InvokeArgs and no Inputs for invokes

### PR DESCRIPTION
**Pull request by mikhailshilkov**
Part of the fix of https://github.com/pulumi/pulumi-aws/issues/800 relying on https://github.com/pulumi/pulumi/pull/3544
Any usage of this code in SDKs would require to update the Pulumi dependency to 1.6.1+. Downstream tests fail because of that.

### CODE REVIEW

**Comment by mikhailshilkov** [comment]
cc @pgavlin @CyrusNajmabadi

File: pkg/tfgen/generate_csharp.go [hunk 1]
```diff
@@ -1061,7 +1066,7 @@ func (rg *csharpResourceGenerator) generateInputTypes(nts []*csharpNestedType) {
 	// Write each input type.
 	for _, nt := range nts {
 		contract.Assert(nt.isInput)
-		rg.generateInputType(nt.typ, true /*nested*/)
+		rg.generateInputType(nt.typ, "ResourceArgs", true /*nested*/, true /*wrapInput*/)
```
Comment (pgavlin) [inline pkg/tfgen/generate_csharp.go]: Should this always be `ResourceArgs`? We do generate nested types for invokes as well.

Comment (mikhailshilkov) [reply pkg/tfgen/generate_csharp.go]: Thanks, I found an example of this. I'll change it.

Comment (mikhailshilkov) [reply pkg/tfgen/generate_csharp.go]: Changed now

**Comment by mikhailshilkov** [comment]
Not sure why the bridge tests are failing, while I only changed the gen...

**Comment by pgavlin** [comment]
Did we completely stop running Travis on this repo? Are we only using GH actions?

**Comment by pgavlin** [comment]
@mikhailshilkov the test failure is my fault--I have a PR to fix it here: https://github.com/pulumi/pulumi-terraform-bridge/pull/83

**Comment by mikhailshilkov** [comment]
@pgavlin Thanks for fixing the tests, I rebased on master. Please have another look.

File: pkg/tfgen/generate_csharp.go [hunk 2]
```diff
@@ -1314,3 +1333,12 @@ func csharpDefaultValue(prop *variable) string {
 
 	return defaultValue
 }
+
+func isValueType(typ string) bool {
```
Comment (pgavlin) [inline pkg/tfgen/generate_csharp.go]: nit: can we use a `kind` for the parameter type rather than `string`?

File: pkg/tfgen/generate_csharp.go [hunk 3]
```diff
@@ -64,7 +64,7 @@ type declarer interface {
 type csharpNestedType struct {
 	declarer declarer
 	isInput  bool
-	isPlain  bool
+	isInvoke bool
```
Comment (pgavlin) [inline pkg/tfgen/generate_csharp.go]: Was this change necessary? `isPlain` was intended to indicate that the type did not need to be marked up with `[Output]` attributes; it seems like it could serve the same purpose. To that end, I'd love to see a diff for the `Config.cs` files with these changes.

Comment (mikhailshilkov) [reply pkg/tfgen/generate_csharp.go]: I did not find any usage of `isPlain` other than passing it around. What about `Config.cs` that had to change? My test run on Azure doesn't yield any.

File: pkg/tfgen/generate_csharp.go (hunk 2 above)
Comment (mikhailshilkov) [reply pkg/tfgen/generate_csharp.go]: I think I started with that, but then `isValueType` should also accept `wrapInput` to understand if it's nullable. It looked like a complication.

**Comment by mikhailshilkov** [comment]
@pgavlin We can't update to the newest Pulumi versions without this fix. Let's agree on what still needs to happen here.
